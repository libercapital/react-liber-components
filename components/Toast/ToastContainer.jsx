import React, { Component } from 'react';

class ToastContainer extends Component {
  render() {
    return <div />;
  }
}

export default ToastContainer;
