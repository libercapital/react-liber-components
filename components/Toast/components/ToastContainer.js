"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactRedux = require("react-redux");

var _ToastList = _interopRequireDefault(require("./ToastList"));

var _store = _interopRequireDefault(require("../store"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = function _default(props) {
  return _react["default"].createElement(_reactRedux.Provider, {
    store: _store["default"]
  }, _react["default"].createElement(_ToastList["default"], props));
};

exports["default"] = _default;