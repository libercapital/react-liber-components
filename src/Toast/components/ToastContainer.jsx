import React from 'react';
import { Provider } from 'react-redux';
import ToastList from './ToastList';
import store from '../store';

export default props => (
  <Provider store={store}>
    <ToastList {...props} />
  </Provider>
);
