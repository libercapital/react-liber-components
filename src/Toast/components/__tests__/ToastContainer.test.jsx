import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import ToastContainer from '../ToastContainer';

describe('Testing ToastContainer Component', () => {
  let props;
  configure({ adapter: new Adapter() });
  beforeEach(() => {
    props = {
      toasts: [],
      close: sinon.spy(),
      remove: sinon.spy(),
    };
  });
  it('should render correctly ', () => {
    const wrapper = shallow(<ToastContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
