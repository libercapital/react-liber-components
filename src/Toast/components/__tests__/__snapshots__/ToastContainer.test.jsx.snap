// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ToastContainer Component should render correctly  1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ToastList)
    close={[Function]}
    remove={[Function]}
    toasts={Array []}
  />
</Provider>
`;
